core/arch/arm/plat-imx/config/config_imx6sx.h

```diff
@@ -19,12 +19,23 @@
 #define DRAM0_SIZE		CFG_DDR_SIZE
 
 /* Location of trusted dram */
-#define TZDRAM_BASE		(DRAM0_BASE - 32 * 1024 * 1024 + CFG_DDR_SIZE)
-#define TZDRAM_SIZE		(30 * 1024 * 1024)
+#ifndef CFG_TZDRAM_BASE
+#define CFG_TZDRAM_BASE		(DRAM0_BASE - 32 * 1024 * 1024 + CFG_DDR_SIZE)
+#endif
+#ifndef CFG_TZDRAM_SIZE
+#define CFG_TZDRAM_SIZE		(30 * 1024 * 1024)
+#endif
+
+#define TZDRAM_BASE		CFG_TZDRAM_BASE
```

Comment:
It's a bit odd that other memory layout options like `CFG_SHMEM_BASE`, `CFG_TEE_LOAD_ADDRESS`, and `CFG_TEE_RAM_START` are configuration options, but `TZDRAM_BASE`, from which the other values are computed, is not.

Is there a reason why `TZDRAM_BASE` is not a CFG option? Some other alternatives:
 - Change all occurrences of `TZDRAM_BASE` with `CFG_TZDRAM_BASE`
 - Remove `TZDRAM_BASE` and supply values for `CFG_TEE_RAM_START` instead

Reply:
Hi @jordanrh1,

No particular reason other than historical I think (some defines predate the introduction of the CFG_ system, at that time it was cumbersome to define some variable in a Makefile and later use it in C code).

Reply:
Current platform defines for `TZDRAM_BASE`:
```

plat-marvell/platform_config.h:  #define TZDRAM_BASE    0x04400000
plat-marvell/platform_config.h:  #define TZDRAM_BASE    0x04400000
plat-ls/platform_config.h:       #define TZDRAM_BASE    (CFG_DDR_TEETZ_RESERVED_START)
plat-sprd/platform_config.h:     #define TZDRAM_BASE    0x8f600000
plat-poplar/platform_config.h:   #define TZDRAM_BASE    0x03200000
plat-poplar/platform_config.h:   #define TZDRAM_BASE    0x03000000
plat-zynqmp/platform_config.h:   #define TZDRAM_BASE    0x60000000
plat-hikey/platform_config.h:    #define TZDRAM_BASE    0x3F200000
plat-hikey/platform_config.h:    #define TZDRAM_BASE    0x3F000000
plat-mediatek/platform_config.h: #define TZDRAM_BASE    0xBE000000
plat-ti/platform_config.h:       #define TZDRAM_BASE    0xbdb00000
plat-ti/platform_config.h:       #define TZDRAM_BASE    0xbdb00000
plat-sam/platform_config.h:      #define TZDRAM_BASE    0x30000000
plat-imx/config/config_imx7.h:   #define TZDRAM_BASE    (DRAM0_BASE + CFG_DDR_SIZE - 32 * 1024 * 1024)
plat-imx/config/config_imx6sx.h: #define TZDRAM_BASE    (DRAM0_BASE - 32 * 1024 * 1024 + CFG_DDR_SIZE)
plat-imx/platform_config.h:      #define TZDRAM_BASE    (CFG_DDR_TEETZ_RESERVED_START)
plat-imx/platform_config.h:      #define TZDRAM_BASE    (CFG_DDR_TEETZ_RESERVED_START)
plat-imx/platform_config.h:      #define TZDRAM_BASE    (CFG_DDR_TEETZ_RESERVED_START)
plat-rcar/platform_config.h:     #define TZDRAM_BASE    0x44000000
plat-rockchip/platform_config.h: #define TZDRAM_BASE    CFG_TZDRAM_RSV_START
plat-d02/platform_config.h:      #define TZDRAM_BASE    0x50460000
plat-d02/platform_config.h:      #define TZDRAM_BASE    0x50400000
plat-stm/platform_config.h:      #define TZDRAM_BASE    CFG_DDR_TEETZ_RESERVED_START
plat-stm/platform_config.h:      #define TZDRAM_BASE    CFG_DDR_TEETZ_RESERVED_START
plat-zynq7k/platform_config.h:   #define TZDRAM_BASE    (CFG_DDR_TEETZ_RESERVED_START)
plat-zynq7k/platform_config.h:   #define TZDRAM_BASE    (CFG_DDR_TEETZ_RESERVED_START)
plat-rpi3/platform_config.h:     #define TZDRAM_BASE    (CFG_SHMEM_START + CFG_SHMEM_SIZE)
```
Maybe you could use `CFG_DDR_TEETZ_RESERVED_START`/`_SIZE`  instead of a new `CFG_TZDRAM_BASE`/`_SIZE`.

Reply:
`CFG_DDR_TEETZ_RESERVED_START` is the nearest precedent there is, but it seems like an unnecessary layer on top of the other CFG options. It's not used in C code anywhere so it's just a convention with no real enforcement of it's usage. I think the cleaner options are to
 - Rename `TZDRAM_BASE` to `CFG_TZDRAM_BASE` thereby making it configurable
 - Remove `TZDRAM_BASE`, `_SIZE` and instead use `CFG_TEE_RAM_START`, `_PH_SIZE`, `_VA_SIZE`.

The following line in `./core/arch/arm/mm/core_mmu.c` would change from:
```
static struct memaccess_area secure_only[] = {
#ifdef TZSRAM_BASE
        MEMACCESS_AREA(TZSRAM_BASE, TZSRAM_SIZE),
#endif
        MEMACCESS_AREA(TZDRAM_BASE, TZDRAM_SIZE),
};
```
to:
```
static struct memaccess_area secure_only[] = {
#ifdef TZSRAM_BASE
        MEMACCESS_AREA(TZSRAM_BASE, TZSRAM_SIZE),
#endif
    MEMACCESS_AREA(CFG_TEE_RAM_START, CFG_TEE_RAM_VA_SIZE),
    MEMACCESS_AREA(CFG_TA_RAM_START, CFG_TA_RAM_SIZE),
};
```

Reply:
> CFG_DDR_TEETZ_RESERVED_START is the nearest precedent there is, but it seems like an unnecessary layer on top of the other CFG options. It's not used in C code anywhere so it's just a convention with no real enforcement of it's usage. 

Fully true.

For the proposal, 2 comments.
* Secure memory could be more than TEE_RAM and TA_RAM, since used in `core_pbuf_is(CORE_MEM_SEC, paddr_t pbuf, size_t len)`.
* `TZSRAM_BASE` and `CFG_TEE_RAM_START` will overlap on some configurations. Maybe it is not an issue. To check.

It could be nice to have something simple. But current platform ways must be maintained.

Reply:
It seems like the least disruptive option is to rename `TZDRAM_BASE` to `CFG_TZDRAM_SIZE`, which makes the value configurable without adding or removing any options. The pull request is split into two commits:
1. rename `TZDRAM_BASE _SIZE` to `CFG_TZDRAM_BASE _SIZE` with no functional change. The rename should not break any external contract because `TZDRAM_BASE` is not currently configurable. 
2. allow `CFG_TZDRAM_*` and `CFG_SHMEM_*` to be specified externally on IMX platforms. This does not break any external contract because `CFG_DDR_TEETZ_RESERVED_START` is not currently configurable.